backend/Auth/ResetPassword.ts

```diff
@@ -64,3 +68,50 @@ router.post(
     return res.status(204).end();
   }
 );
+
+router.get(
+  "/validate-token/:token",
+  param("token").exists(),
+  async (req, res) => {
+    const errors = validationResult(req);
+    if (!errors.isEmpty()) {
+      return invalidRequest(res, errors);
+    }
+    const token = req.params?.token;
```

Comment:
My understanding is that `req.params` is always defined, so the `?.` is not necessary here. You can even get typescript to be aware of what params and query params are available by providing generics when calling `router.get()`.

If the reason you used the optional chaining operator is to get typescript to be okay, the better solution is just to strictly type the params. See the code below as an example:

https://github.com/JustUtahCoders/flax-crm/blob/9abb1948c270a35475106c445383123bfb2d83b3/backend/Fields/BatchPostFields.ts#L7

Reply:
Thank you!  Passing generics for params, fixed the typescript errors.